OpenRA.Game/Player.cs

```diff
@@ -267,15 +267,15 @@ public bool IsAlliedWith(Player p)
 
 		public Color PlayerRelationshipColor(Actor a)
 		{
-			var player = a.World.RenderPlayer ?? a.World.LocalPlayer;
+			var renderPlayer = a.World.RenderPlayer;
+			var player = renderPlayer ?? a.World.LocalPlayer;
 			if (player != null && !player.Spectating)
 			{
-				var apparentOwner = a.EffectiveOwner != null && a.EffectiveOwner.Disguised
-					? a.EffectiveOwner.Owner
-					: a.Owner;
-
-				if (a.Owner.IsAlliedWith(a.World.RenderPlayer))
-					apparentOwner = a.Owner;
+				var effectiveOwner = a.EffectiveOwner;
+				var apparentOwner = a.Owner;
+				if (effectiveOwner != null && effectiveOwner.Disguised)
+					if (!a.Owner.IsAlliedWith(renderPlayer))
```

Comment:
In my opinion you might as well merge the `if`s, I don't think the nesting improves readability much here.